actions/actions.py

```diff
@@ -222,7 +224,11 @@ def name(self) -> Text:
         return "action_set_faq_slot"
 
     def run(self, dispatcher, tracker, domain) -> List[EventType]:
-        fullintent = tracker.latest_message["intent"].get("full_retrieval_intent")
+        fullintent = (
+            tracker.latest_message.get("response_selector", {})
+            .get("faq", {})
+            .get("full_retrieval_intent")
+        )
         if fullintent and fullintent.startswith("faq"):
             topic = fullintent.split("/")[1]
```

Comment:
```suggestion
        full_intent = (
            tracker.latest_message.get("response_selector", {})
            .get("faq", {})
            .get("full_retrieval_intent")
        )
        if full_intent and full_intent.startswith("faq"):
            topic = full_intent.split("/")[1]
```
also don't think we need the `startswith` check now that you've changed it to get from `faq`

Reply:
Ah this is true

Reply:
```suggestion
        full_intent = (
            tracker.latest_message.get("response_selector", {})
            .get("faq", {})
            .get("full_retrieval_intent")
        )
        if full_intent:
            topic = full_intent.split("/")[1]
```